compiler/src/dmd/common/file.d

```diff
@@ -14,15 +14,23 @@
 
 module dmd.common.file;
 
+import core.stdc.stdio;
+import core.stdc.stdlib;
+import core.stdc.string;
+import core.stdc.limits;
+
 import core.stdc.errno : errno;
 import core.stdc.stdio : fprintf, remove, rename, stderr;
 import core.stdc.stdlib;
 import core.stdc.string : strerror, strlen, memcpy;
 
 import dmd.common.smallbuffer;
+import dmd.root.filename;
+import dmd.root.rmem;
```

Comment:
Historically, dmd.common was added because the backend may not depend on dmd.root. Is that restriction now lifted? In that case we could merge dmd.common and dmd.root.